test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTest.java

```diff
@@ -0,0 +1,267 @@
+        Assert.assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Verify segment stores is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("segmentstore", zkUri, conURIs.get(0), 2, 0.1, 1000.0);
+        Assert.assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega segment store instance details: {}", segmentStoreInstance.getServiceDetails());
+    }
+
+    @Test(timeout = 600000)
+    public void multiReaderWriterTest() throws InterruptedException, URISyntaxException, ExecutionException  {
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest();
+
+        //scale down SSS by 1 instance
+        segmentStoreInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 SSS instance down");
+        readWriteTest();
+
+        segmentStoreInstance.scaleService(2, true);
+        Thread.sleep(60000);
+        controllerInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 controller instance down");
+        readWriteTest();
+
+        segmentStoreInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 controller  and 1 SSS instance down");
+        readWriteTest();
+    }
+
+    private void readWriteTest() throws InterruptedException, ExecutionException {
+
+        String scope = "testMultiReaderWriterScope" + new Random().nextInt(Integer.MAX_VALUE);
+        String readerGroupName = "testMultiReaderWriterReaderGroup" + new Random().nextInt(Integer.MAX_VALUE);
+
+        //20  readers -> 20 stream segments ( to have max read parallelism)
+        ScalingPolicy scalingPolicy = ScalingPolicy.fixed(5);
+        StreamConfiguration config = StreamConfiguration.builder().scope(scope)
+                .streamName(STREAM_NAME).scalingPolicy(scalingPolicy).build();
+
+        URI controllerUri = getControllerURI();
+        Controller controller = getController(controllerUri);
+
+        //create a scope
+        Boolean createScopeStatus = controller.createScope(scope).get();
+        log.debug("Create scope status {}", createScopeStatus);
+
+        //create a stream
+        Boolean createStreamStatus = controller.createStream(config).get();
+        log.debug("Create stream status {}", createStreamStatus);
+         eventsReadFromPravega = new ConcurrentLinkedQueue<>();
+
+         //stopWriteFlag = new AtomicBoolean(false);
+         stopReadFlag = new AtomicBoolean(false);
+         eventData = new AtomicLong(); //data used by each of the writers.
+         eventReadCount = new AtomicLong(); // used by readers to maintain a count of events.
+
+        log.info("events read from pravega {}", eventsReadFromPravega);
```

Comment:
it is empty here, we could remove this log statement . Formatting also needs to be updated.